app/models/master_file.rb

```diff
@@ -352,6 +357,12 @@ def update_derivatives(outputs, managed = true)
     save
   end
 
+  def add_supplemental_files(ids)
+    self.supplemental_files += SupplementalFile.find(ids)
+
+    save
```

Comment:
Could the save be moved up into `update_progress_on_success!` To avoid saving both here and in `update_derivatives`?